owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt

```diff
@@ -195,4 +195,45 @@ class RemoteShare : Parcelable, Serializable {
             }
         }
     }
+
+
+}
+
+/**
+ * // TODO This type is already included in the domain but we still need it here since the parsing takes place in this library for the moment
+ *
+ * Enum for Share Type, with values:
+ * -1 - No shared
+ * 0 - Shared by user
+ * 1 - Shared by group
+ * 3 - Shared by public link
+ * 4 - Shared by e-mail
+ * 5 - Shared by contact
+ * 6 - Federated
+ *
+ * @author masensio
+ */
+enum class ShareType constructor(val value: Int) {
+    NO_SHARED(-1),
+    USER(0),
+    GROUP(1),
+    PUBLIC_LINK(3),
+    EMAIL(4),
+    CONTACT(5),
+    FEDERATED(6);
+
+    companion object {
+        fun fromValue(value: Int): ShareType? {
+            return when (value) {
+                -1 -> NO_SHARED
+                0 -> USER
+                1 -> GROUP
+                3 -> PUBLIC_LINK
+                4 -> EMAIL
+                5 -> CONTACT
+                6 -> FEDERATED
+                else -> null
```

Comment:
Which is the diference between no shared and null in terms of use? Is it needed to have nullability here?

Reply:
This comes from time ago, we can do it better, I'm on it

Reply:
Well, I've just pushed a commit cleaning up `RemoteShare.kt` and `RemoteCapability.kt` but we can not get rid of the null cases because `ShareType` is an enum in which `NO_SHARED` has -1 value so if we call the method `fun fromValue(value: Int): ShareType?` with value -1, it should retrieve `NO_SHARED` type.

Besides, we use `NO_SHARED` to initialise the share type of a Remote Share